chromium_src/chrome/browser/ui/singleton_tabs.cc

```diff
@@ -13,3 +13,10 @@ void ShowSingletonTab(Browser* browser, const GURL& url) {
 
   ShowSingletonTab_ChromiumImpl(browser, new_url);
 }
+
+void ShowSingletonTab(Profile* profile, const GURL& url) {
+  GURL new_url =
```

Comment:
what is going on here? This needs some explanation

Reply:
cc: @cdesouza-chromium. Looks like we already override `void ShowSingletonTab(Browser* browser, const GURL& url)` in this file above, but now upstream added an overload that takes `Profile*` instead of `Browser*` so we are adding another override that does the same thing.

Reply:
Without this override we are gonna have a build breakage, as this overload in particular would get renamed to `  ShowSingletonTab_ChromiumImpl` as well, which breaks upstream code. Linking error example:

```
ld64.lld: error: undefined symbol: ShowSingletonTab(Profile*, GURL const&)
>>> referenced by chrome_pages.cc:177 (../../../src/chrome/browser/ui/chrome_pages.cc:177)
>>>               obj/chrome/browser/ui/libui.a(chrome_pages.o):(symbol chrome::(anonymous namespace)::ShowHelpImpl(Browser*, Profile*, chrome::HelpSource)+0x29d)
```

Reply:
please check git blame and add comments in a follow-up